mantidproject/mantid
# Workspace displays status bar

**Description of work.**
Adds a status bar to the workspace display. This will display messages regarding the status of user actions - e.g. copy completed, operation in progress.

All of that functionality has been refactored into common classes:
- `status_bar_view.py` provides the parent window that contains the status bar.
- `user_notifier.py` provides the functions for notifying the user via tooltip **(Windows only)**, and status bar
- `data_copier.py` provides functions for copying the data from the matrix/table workspaces

- Note that the code in `data_copier.py` and  `user_notifier.py` is copied from the old `table_copying.py`(which has now been deleted), and turned into two classes with unit tests 🎉 .
- Workspace displays are now in `mantidqt.widgets.workspacedisplay` package

**To test:**
- Copy something (bin, spectra or selected cells), the status bar should show "Copy Successful"
- Open a workspace with a lot of spectra and bins, select the _bins_, and copy them. The status bar will show "Operation in progress" if you have selected enough bins (as copying bins is slower than spectra)



Fixes #24630 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
